### ISSUE TITLE
dynamic masking for RoBERTa model

### ISSUE THREAD

**Issue by mingyang3github** [huggingface/transformers#5979]
# ❓ Questions & Help

I saw that "dynamic masking" was mentioned on the README file for language modeling:
"In accordance to the RoBERTa paper, we use dynamic masking rather than static masking. The model may, therefore, converge slightly slower (over-fitting takes more epochs)."

I couldn't find which class this method is implemented in and how to enable this feature during pre-training using the Trainer class. Could someone please help me?

Thank you very much in advance.

**Comment by patil-suraj** [comment]
Hi @mingyang3github 
masking is implemented in ` DataCollatorForLanguageModeling`'s `mask_tokens` method, [here](https://github.com/huggingface/transformers/blob/master/src/transformers/data/data_collator.py#L103)

For pre-training you can use the language modeling script, it takes of the dataset and masking.

**Comment by mingyang3github** [comment]
> Hi @mingyang3github
> masking is implemented in ` DataCollatorForLanguageModeling`'s `mask_tokens` method, [here](https://github.com/huggingface/transformers/blob/master/src/transformers/data/data_collator.py#L103)
> 
> For pre-training you can use the language modeling script, it takes of the dataset and masking.

Hi @patil-suraj ,
Thank you for your reply. If I understand this correctly, DataCollatorForLanguageModeling's method is static masking, right?

Based on RoBERTa model's paper, the dynamic masking refers to "training data was duplicated 10 times so that each sequence is masked in 10 different ways over the 40 epochs of training."
Has this method implemented been implemented in DataCollatorForLanguageModeling ?

**Comment by patil-suraj** [comment]
It is dynamic masking, as the masking is handled in the collate function, the same examples gets masked differently. Because every batch first goes through collate function before the batch returned from the loader, so each time the examples goes through collate it gets masked differently.

**Comment by buaapengbo** [comment]
Hi @patil-suraj , is that mean we **always use dynamic masking**,  when use `run_language_modeling.py`, no matter pre-training for Bert, RoBERTa or else.

**Comment by patil-suraj** [comment]
@buaapengbo
Yes, the `DataCollatorForLanguageModeling` always does dynamic masking no matter the model

**Comment by mingyang3github** [comment]
I thought I responded, but it turns out I didn't. Sorry.
@patil-suraj Thank you very much! That makes sense!

**Comment by HaishuoFang** [comment]
Thanks for the answer.  For the original roberta, they seem to use the same mask per sentence 4 times during training and 10 different masks for the same sentence during the whole training. But this function always returns different masks for the same sentence during training. This means the model always receives different masks for the same sentence during the whole training process right?
